scipy/stats/_stats_py.py

```diff
@@ -9649,6 +9650,259 @@ def combine_pvalues(pvalues, method='fisher', weights=None):
     return SignificanceResult(statistic, pval)
 
 
+QuantileTestResultBase = _make_tuple_bunch('QuantileTestResultBase',
+                                           ['statistic1', 'statistic2', 'pvalue'], [])
+
+
+class QuantileTestResult(QuantileTestResultBase):
+    r"""
+    Result of `scipy.stats.quantile_test`.
+
+    Attributes
+    ----------
+    statistic: float
+        The observed proportion of values less than or equal to the
+        hypothesized quantile.
+    pvalue : float
+        The p-value of the hypothesis test.
+
+
+    """
+
+    def __init__(self, statistic1, statistic2, pvalue, alternative, x, p):
+        super().__init__(statistic1, statistic2, pvalue)    
+        self._alternative = alternative
+        self._x = np.sort(x)
+        self._p = p
+
+    def confidence_interval(self, confidence_level=0.95):
+        """
+        Compute the confidence interval of the quantile.
+
+        Parameters
+        ----------
+        confidence_level : float, default: 0.95
+            Confidence level for the computed confidence interval
+            of the quantile. Default is 0.95.
+
+        Returns
+        -------
+        ci : ``ConfidenceInterval`` object
+            The object has attributes ``low`` and ``high`` that hold the
+            lower and upper bounds of the confidence interval.
+
+        References
+        ----------
+
+        Examples
+        --------
+
+        """
+        alternative = self._alternative
+        p = self._p
+        x = np.sort(self._x)
+        n = len(x)
+        bd = stats.binom(n, p)
+
+        if confidence_level < 0 or confidence_level > 1:
+            message = "`confidence_level` must be a number between 0 and 1."
+            raise ValueError(message)
+
+        low_index = np.nan
+        high_index = np.nan
+
+        if alternative == 'less':
+            p = 1 - confidence_level
+            low = -np.inf
+            high_index = int(bd.isf(p))
+            high = x[high_index] if high_index < n else np.nan
+        elif alternative == 'greater':
+            p = 1 - confidence_level
+            low_index = int(bd.ppf(p)) - 1
+            low = x[low_index] if low_index >= 0 else np.nan
+            high = np.inf
+        elif alternative == 'two-sided':
+            p = (1 - confidence_level) / 2
+            low_index = int(bd.ppf(p)) - 1
+            low = x[low_index] if low_index >= 0 else np.nan
+            high_index = int(bd.isf(p))
+            high = x[high_index] if high_index < n else np.nan
+
+        return ConfidenceInterval(low, high)
+
+
+def quantile_test_iv(x, q, p, alternative):
```

Comment:
I don‘t know the rest of the code base, but the function name seems cryptic, in particular without a docstring. I guess iv stands for input validation.

Reply:
This is fine, we use this convention elsewhere. Hence familiar to stats maintainers 😃